src/main/java/org/broadinstitute/dsm/route/FilterRoute.java

```diff
@@ -238,22 +244,30 @@ else if (savedFilters != null) {
 
     public static List<?> filterParticipantList(Filter[] filters, Map<String, DBElement> columnNameMap, @NonNull DDPInstance instance) {
         Map<String, String> queryConditions = new HashMap<>();
+        List<ParticipantDataDto> allParticipantData = participantDataDao
+                .getParticipantDataByInstanceid(Integer.parseInt(instance.getDdpInstanceId()));
```

Comment:
I wonder if we really should have that here. If there is not filter for PARTICIPANT_DATA.equals(filter.getParticipantColumn().tableAlias) then we wouldn't need the list of participantData…

Reply:
Yes, at first I had it inside that if block you mentioned, but in this case we would need to do it for each filter with participantData and that's why moved it there. But I guess there won't be too many filters so maybe it was better to leave it inside the if block not to be called in every case. So do you think I should move it into the if block?